application/ui/components/input/array.jsx

```diff
@@ -0,0 +1,147 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _             = require('underscore');
+var React         = require('react');
+
+module.exports = React.createClass({
+
+    displayName : 'ArrayParameter',
+
+    getInitialState : function()
+    {
+        return {
+            values   : [],
+            numItems : 0
```

Comment:
Now that I look closer in order to understand your comment, it occurs to me that `numItems` isn't actually getting used anywhere. Am I wrong? We're incrementing it when a field is added but not actually using it anywhere else.